### PR TITLE
Ins 19 node index

### PR DESCRIPTION
**Pull request by LinnikD**
**- What I did**

Index on node public key

### CODE REVIEW

File: application/contract/member/member.go [hunk 1]
```diff
@@ -195,3 +197,24 @@ func (m *Member) RegisterNodeCall(ref core.RecordRef, params []byte) (interface{
...
+	rootDomain := rootdomain.GetObject(ref)
+	nodeDomainRef, err := rootDomain.GetNodeDomainRef()
+	if err != nil {
+		return nil, fmt.Errorf("[ getNodeRef ] %s", err.Error())
+	}
+
+	nd := nodedomain.GetObject(nodeDomainRef)
+	nodeRef, err := nd.GetNodeRefByPK(publicKey)
+	if err != nil {
+		return nil, fmt.Errorf("[ getNodeRef ] %s", err.Error())
```
Comment (megge-dream) [inline application/contract/member/member.go]: Use different message in error

File: functest/register_node_test.go [hunk 2]
```diff
@@ -117,3 +126,36 @@ func TestReceiveNodeCert(t *testing.T) {
 		})
 	}
 }
+
+func TestGetNodeRefByPK(t *testing.T) {
```
Comment (megge-dream) [inline functest/register_node_test.go]: Move all tests about GetNodeRef to separate file

File: application/contract/nodedomain/nodedomain.go [hunk 3]
```diff
@@ -56,11 +60,29 @@ func (nd *NodeDomain) RegisterNode(publicKey string, role string) (string, error
...
+	}
+	return nodeRef, nil
 }
 
 // RemoveNode deletes node from registry
 func (nd *NodeDomain) RemoveNode(nodeRef core.RecordRef) error {
 	node := nd.getNodeRecord(nodeRef)
+	nodePK, err := node.GetPublicKey()
+	if err != nil {
+		return err
```
Comment (ivanshibitov) [inline application/contract/nodedomain/nodedomain.go]: wrap

File: application/contract/nodedomain/nodedomain.go [hunk 4]
```diff
@@ -56,11 +60,29 @@ func (nd *NodeDomain) RegisterNode(publicKey string, role string) (string, error
...
+	nd.NodeIndexPK[publicKey] = newNodeRef
+
+	return newNodeRef, err
+}
+
+// GetNodeRefByPK returns node ref
+func (nd *NodeDomain) GetNodeRefByPK(publicKey string) (string, error) {
+	nodeRef, ok := nd.NodeIndexPK[publicKey]
+	if !ok {
+		return nodeRef, fmt.Errorf("[ GetNodeRefByPK ] Node not found by PK: %s", publicKey)
```
Comment (ivanshibitov) [inline application/contract/nodedomain/nodedomain.go]: wrap

**Pull request merged**